ste/sender-blobFolders.go

```diff
@@ -53,6 +54,53 @@ func newBlobFolderSender(jptm IJobPartTransferMgr, destination string, p pipelin
 	return out, nil
 }
 
+func (b *blobFolderSender) overwriteDFSProperties() (string, error) {
+	b.jptm.Log(pipeline.LogWarning, "It is impossible to completely overwrite a folder with existing content under it on a hierarchical namespace storage account. A best-effort attempt will be made, but if CPK does not match the transfer will fail.")
+
+	b.metadataToApply["hdi_isfolder"] = "true" // Set folder metadata flag
+	err := b.getExtraProperties()
+	if err != nil {
+		return "Get Extra Properties", fmt.Errorf("when getting additional folder properties: %w", err)
+	}
+
+	// SetMetadata can set CPK if it wasn't specified prior. This is not a "full" overwrite, but a best-effort overwrite.
+	_, err = b.destination.SetMetadata(b.jptm.Context(), b.metadataToApply, azblob.BlobAccessConditions{}, b.cpkToApply)
+	if err != nil {
+		return "Set Metadata", fmt.Errorf("A best-effort overwrite was attempted; CPK errors cannot be handled when the blob cannot be deleted.\n%w", err)
+	}
+
+	_, err = b.destination.SetTags(b.jptm.Context(), nil, nil, nil, b.blobTagsToApply)
+	if err != nil {
+		return "Set Blob Tags", err
+	}
+	_, err = b.destination.SetHTTPHeaders(b.jptm.Context(), b.headersToAppply, azblob.BlobAccessConditions{})
+	if err != nil {
+		return "Set HTTP Headers", err
+	}
+
+	// Upload ADLS Gen 2 ACLs
+	if b.jptm.FromTo() == common.EFromTo.BlobBlob() && b.jptm.Info().PreserveSMBPermissions.IsTruthy() {
```

Comment:
how do we know if the destination is adls?

Reply:
The only place this is called is when we receive an HNS specific error message.